src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -219,7 +113,7 @@ public JsonConverter GetConverter(Type typeToConvert)
                 ThrowHelper.ThrowArgumentNullException(nameof(typeToConvert));
             }
 
-            RootReflectionSerializerDependencies();
+            EnsureInitializedForReflectionSerializer();
```

Comment:
Rooting the reflection dependencies is a subset of initializing for the reflection serialization. This is now setting the `IsInitializedForReflectionSerializer` flag when it wasn't doing that in the past, which can result in subtle breaking changes (e.g. when a user calls `GetConverter` in an `Options` instance tied to a `JsonSerializerContext`).